systems/common/msgBusServiceClient/msgbus_service_client.go

```diff
@@ -0,0 +1,151 @@
+	uuid         string
+	service      string
+	system       string
+	instanceId   string
+	msgBusURI    string
+	msgClientURI string
+	exchange     string
+	listQueue    string
+	publQueue    string
+	conn         *grpc.ClientConn
+	client       pb.MsgClientServiceClient
+	timeout      time.Duration
+	host         string
+	retry        int8
+	routes       []string
+}
+
+func NewMsgBusClient(timeout time.Duration, system string,
+	service string, instanceId string, msgBusURI string,
+	serviceURI string, msgClientURI string, exchange string, lq string, pq string, retry int8, routes []string) *msgBusServiceClient {
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, msgClientURI, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		logrus.Fatalf("did not connect: %v", err)
+	}
+	client := pb.NewMsgClientServiceClient(conn)
+
+	return &msgBusServiceClient{
+		service:      service,
+		system:       system,
+		instanceId:   instanceId,
+		msgBusURI:    msgBusURI,
+		msgClientURI: msgClientURI,
+		conn:         conn,
+		client:       client,
+		timeout:      timeout,
+		retry:        retry,
+		host:         serviceURI,
+		routes:       routes,
+		listQueue:    lq,
+		publQueue:    pq,
+		exchange:     exchange,
+	}
+
+}
+
+func (m *msgBusServiceClient) Register() error {
+	logrus.Debugf("Registering %s service instance %s to MessageBusClient at %s.", m.service, m.instanceId, m.msgClientURI)
+	ctx, cancel := context.WithTimeout(context.Background(), m.timeout)
+	defer cancel()
+
+	resp, err := m.client.RegisterService(ctx, &pb.RegisterServiceReq{
+		ServiceName: m.service,
+		SystemName:  m.system,
+		MsgBusURI:   m.msgBusURI,
+		ServiceURI:  m.host,
+		InstanceId:  m.instanceId,
+		Routes:      m.routes,
+		ListQueue:   m.listQueue,
+		PublQueue:   m.publQueue,
+		Exchange:    m.exchange,
+		GrpcTimeout: uint32(m.timeout.Seconds())})
+	if err != nil {
+		return err
+	}
+
+	if resp.GetState() == pb.REGISTRAION_STATUS_REGISTERED {
+		m.uuid = resp.ServiceUuid
+	}
+
+	logrus.Infof("%s service instance %s to MessageBusClient at %s.", m.service, m.instanceId, resp.State.String())
+	return nil
+}
+
+func (m *msgBusServiceClient) Start() error {
+	logrus.Debugf("Starting MessageClientRoutine for %s service instance %s Routine ID %s.", m.service, m.instanceId, m.uuid)
+	ctx, cancel := context.WithTimeout(context.Background(), m.timeout)
+	defer cancel()
+
+	_, err := m.client.StartMsgBusHandler(ctx, &pb.StartMsgBusHandlerReq{
+		ServiceUuid: m.uuid})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (m *msgBusServiceClient) Stop() error {
+	logrus.Debugf("Stopping MessageClientRoutine for %s service instance %s Routine ID %s.", m.service, m.instanceId, m.uuid)
+	ctx, cancel := context.WithTimeout(context.Background(), m.timeout)
+	defer cancel()
+
+	_, err := m.client.StopMsgBusHandler(ctx, &pb.StopMsgBusHandlerReq{
+		ServiceUuid: m.uuid,
```

Comment:
Same here m.uuid is assigned before check if it has been successfully generated .